server/artifacts/artifact_server.go

```diff
@@ -95,9 +95,12 @@ func (a *ArtifactServer) gateKeeping(r *http.Request) (context.Context, error) {
 	if token == "" {
 		cookie, err := r.Cookie("authorization")
 		if err != nil {
-			return nil, err
+			if err != http.ErrNoCookie {
+				return nil, err
```

Comment:
For `auth-mode=server`, there is no `authorization` header or cookie.  Will it work?

Reply:
that's what this should fix